bundles/net.tourbook/src/net/tourbook/tour/ActionDeletePausesDialog.java

```diff
@@ -0,0 +1,125 @@
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
+ *******************************************************************************/
+package net.tourbook.tour;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import net.tourbook.Images;
+import net.tourbook.Messages;
+import net.tourbook.application.TourbookPlugin;
+import net.tourbook.common.UI;
+import net.tourbook.data.TourData;
+import net.tourbook.ui.ITourProvider;
+
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.widgets.Display;
+
+public class ActionDeletePausesDialog extends Action {
+
+   private ITourProvider _tourProvider;
+   private int[]         _tourPausesViewSelectedIndices;
+
+   public ActionDeletePausesDialog(final ITourProvider tourProvider) {
+
+      _tourProvider = tourProvider;
+
+      setText(Messages.App_Action_DeleteTourPauses);
+      setImageDescriptor(TourbookPlugin.getImageDescriptor(Images.App_Delete));
+      setDisabledImageDescriptor(TourbookPlugin.getImageDescriptor(Images.App_Delete_Disabled));
+
+      setEnabled(false);
+   }
+
+   private void doAction(final ITourProvider tourProvider) {
+
+      final ArrayList<TourData> selectedTours = tourProvider.getSelectedTours();
+
+      // check that only one tour is selected
+      if (selectedTours == null || selectedTours.size() != 1 || selectedTours.get(0) == null) {
+         return;
+      }
+
+      final TourData tourData = selectedTours.get(0);
+
+      if (tourData.isManualTour()) {
+         // a manually created tour do not have time slices -> no  pauses
+         return;
+      }
+
+      final String dialogTitle = Messages.Dialog_DeleteTourPauses_Title;
+
+      final String[] incrementedSelectedIndices = Arrays.stream(_tourPausesViewSelectedIndices)
+            .map(index -> ++index)
+            .mapToObj(String::valueOf)
+            .toArray(String[]::new);
+      final String dialogMessage = NLS.bind(Messages.Dialog_DeleteTourPauses_Message, String.join(UI.COMMA_SPACE, incrementedSelectedIndices));
+
+      if (!MessageDialog.openQuestion(
+            Display.getDefault().getActiveShell(),
+            dialogTitle,
+            dialogMessage)) {
+         return;
+      }
+
+      final List<Long> listPausedTime_Start = Arrays.stream(tourData.getPausedTime_Start()).boxed()
+            .collect(Collectors.toList());
+      final List<Long> listPausedTime_End = Arrays.stream(tourData.getPausedTime_End()).boxed()
+            .collect(Collectors.toList());
+      final List<Long> listPausedTime_Data = Arrays.stream(tourData.getPausedTime_Data()).boxed()
+            .collect(Collectors.toList());
```

Comment:
I would be easier to read when these 6 lines are not wrapped, line 74 is already longer

Reply:
Fixed. I typically don't like long line because that requires scrolling in Eclipse but not sure what the code standards are

Reply:
When I use the default window sizes for my Eclipse IDE and MT windows, then these 3 lines are fully visible 

![image](https://github.com/mytourbook/mytourbook/assets/1283445/92c51b4c-7f1e-4a6c-a3fe-66b4376c5a1a)